lib/Twig/Template.php

```diff
@@ -204,6 +204,8 @@ public function displayBlock($name, array $context, array $blocks = array(), $us
             }
         } elseif (false !== $parent = $this->getParent($context)) {
             $parent->displayBlock($name, $context, array_merge($this->blocks, $blocks), false);
+        } elseif (isset($blocks[$name])) {
+            throw new Twig_Error_Runtime(sprintf('Block "%s" should not call parent() in %s as the block does not exist in the parent template %s.', $name, $blocks[$name][0]->getTemplateName(), $this->getTemplateName()), -1, $blocks[$name][0]->getTemplateName());
```

Comment:
Oh, didn't notice that before, but I would enclose the `%s` placeholders for the template names with double quotes for consistency with other exception messages.